src/ordered.jl

```diff
@@ -55,43 +54,39 @@ function colordither(
     # Precompute lookup tables for modulo indexing of threshold matrix
     mat = numerator.(alg.mat)
     nmax = maximum(mat)
-    matsize = size(mat)
-    rlookup = [mod1(i, matsize[1]) for i in 1:size(img)[1]]
-    clookup = [mod1(i, matsize[2]) for i in 1:size(img)[2]]
+    hm, wm = size(mat)
+    rlut = [mod1(i, hm) for i in axes(img, 1)]
+    clut = [mod1(i, wm) for i in axes(img, 2)]
 
     # Allocate matrices
-    candidates = Array{Int}(undef, nmax)
     index = Matrix{Int}(undef, size(img)...)
 
-    @inbounds for I in CartesianIndices(img)
+    Threads.@threads for I in CartesianIndices(img)
```

Comment:
> is there a specific order Threads.@threads, @inbounds and @views are best called in?

No.

BTW, Halide has a very illustrative demo on different parallel strategy 
 that might worth a reading: https://halide-lang.org/tutorials/tutorial_lesson_05_scheduling_1.html